cupy/cusparse.py

```diff
@@ -345,7 +349,7 @@ def csr2dense(x, out=None):
 
     """
     dtype = x.dtype
-    assert dtype == 'f' or dtype == 'd'
+    assert dtype == 'f' or dtype == 'd' or dtype == 'F' or dtype == 'D'
```

Comment:
How about `dtype.char in 'fdFD'`? Maybe simple and fast.

Reply:
That sounds good, I  tested the `in` operation in ipython and find it much faster than original code.
```python
In [1]: a='D'

In [2]: %timeit a=="f" or a=="d" or a=="F" or a=="D"
102 ns ± 4.83 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [3]: %timeit a in "fdFD"
29 ns ± 0.279 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```